### DISCUSSION TITLE
How implement json web token in .net c#?

### DISCUSSION THREAD

**Discussion by aisakk** [dotnet/aspnetcore#53438] [Q&A] [answered]
Hello my friends, i need help, I'm reading this documentation to [implement JWT ](https://learn.microsoft.com/es-es/aspnet/core/security/authentication/jwt-authn?view=aspnetcore-8.0) but the documentation say that this is for development local, but if i want develop the token right from the start or this documentation, Does it help me implement JWT globally?, If so, how would it be implemented?.

I would also like to know if there is documentation about jwt that I can implement in projects or should I just use dotnet new user-jwt? Please, I've been doing this for 3 days and I can't find documentation on how to generate a jwt token.

In a few words, what I want to do is generate a token and pass it through the API so that the frontend can capture it, the backend would be in .net with c#

**Comment by gfoidl** [comment]
You need something that acts as "token provider", see https://jwt.io/introduction
Which token provider you use depends on your scenario, as there are many possibilities, so it's hard to tell a general purpose solution for this.

**Reply by aisakk** [reply]
I see jwt.io and this plataform recommend a package, but im going the page about the package, and not return a example of code how use, I need how implement JWT with .net or c#, because if i want generate a token to pass on the API Minimals, I'm gracefull if help me to generate token.

**Reply by gfoidl** [reply, marked as answer]
Before creating the JWT you need to verify the user's credentials _somehow_.

The [JwtSecurityToken](https://learn.microsoft.com/en-us/dotnet/api/system.identitymodel.tokens.jwt.jwtsecuritytoken?view=msal-web-dotnet-latest) can be created with [JwtSecurityTokenHandler](https://learn.microsoft.com/en-us/dotnet/api/system.identitymodel.tokens.jwt.jwtsecuritytokenhandler?view=msal-web-dotnet-latest).

You can also use something like https://github.com/jwt-dotnet/jwt.
And read through [jwt token factory asp.net core](https://www.google.com/search?q=jwttokenfactory+asp.net+core) to understand what's needed for JWT.

**Reply by aisakk** [reply]
I know that I need the user's credentials, but my question is, will Microsoft have an example of how to generate a token or does it only have dotnet user-jwt and that must be used to generate them in the APIs?, And will the link you gave me allow me to generate a token from scratch? sorry for the inconvenience

**Reply by aisakk** [reply]
Sorry my friend I'm see that the page jwt.io/libraries show example code to implement in .net sorry, i need internet to use and test. Thanks for the all answer sorry.
